code/game/machinery/dance_machine.dm

```diff
@@ -76,6 +76,15 @@
 	else
 		icon_state = "disco0"
 
+/obj/machinery/disco/update_overlays()
+	. = ..()
+	underlays.Cut()
+	set_light(0)
```

Comment:
Isn't it better to do this if it ain't active?

Reply:
I wrote all these procs so that initially all underlays are cut then added back on w/ checks. I believe this is common practice, all the update_icon() procs before tg_appearance PR were written this way. Also the Update_overlays() base does this w/ overlays, im just following the same procedure w/ underlays

Reply:
Oh, I meant to call `set_light(0)` when the disco ball ain't active. Instead of always first setting it to 0. Then deciding if it should be on or not.
The underlays stuff is fine yeah. The best would be to have its own proc. But that is something for the future.